### PR TITLE
fix: allow absolute baseUrl in tsconfig.json

### PR DESCRIPTION
**Pull request by nwalters512**
`configLoader` assumed that `baseUrl` would always be relative path, but the TypeScript compiler allows `baseUrl` to be an absolute path as well. This PR updates the `configLoader` to correctly compute `absoluteBaseUrl` if the `baseUrl` in the TypeScript config file is already absolute.

Note that the same thing was already being done if `explicitParams` was provided.

### CODE REVIEW

File: src/config-loader.ts [hunk 1]
```diff
@@ -85,7 +85,9 @@ export function configLoader({
   }
 
   const tsConfigDir = path.dirname(loadResult.tsConfigPath);
-  const absoluteBaseUrl = path.join(tsConfigDir, loadResult.baseUrl);
+  const absoluteBaseUrl = path.isAbsolute(loadResult.baseUrl)
+    ? loadResult.baseUrl
+    : path.join(tsConfigDir, loadResult.baseUrl);
```
Comment (aleclarson) [inline src/config-loader.ts:90]: Use this instead please:
```ts
const absoluteBaseUrl = path.resolve(tsConfigDir, loadResult.baseUrl);
```

**Review by aleclarson — CHANGES_REQUESTED** [review]

File: src/config-loader.ts (hunk 1 above)
Comment (nwalters512) [inline src/config-loader.ts:90]:
```suggestion
  const absoluteBaseUrl = path.resolve(tsConfigDir, loadResult.baseUrl);
```

**Review by aleclarson — APPROVED** [review]

**Comment by jonaskello** [comment]
@nwalters512 Could you also add an entry for this PR in the CHANGELOG under unreleased heading ? With that it should be good to merge.

**Comment by aleclarson** [comment]
Hey @nwalters512, would you mind adding this to `CHANGELOG.md` and pushing it?

```
## [Unreleased]

### Fixed

- Let `baseUrl` in tsconfig be an absolute path. See PR [#174](https://github.com/dividab/tsconfig-paths/pull/174). Thanks to @nwalters512!
```

**Comment by nwalters512** [comment]
I'm on it! Thanks for getting this reviewed.

**Comment by aleclarson** [comment]
@jonaskello LGTM

**Review by aleclarson — APPROVED** [review]

File: src/__tests__/config-loader.test.ts [hunk 2]
```diff
@@ -97,4 +97,20 @@ describe("config-loader", (): void => {
...
+      // tslint:disable-next-line:no-any
+      tsConfigLoader: (_: any) => ({
+        tsConfigPath: "/baz/tsconfig.json",
+        baseUrl: "/baz",
+        paths: {},
+      }),
+    });
+
+    const successResult = result as ConfigLoaderSuccessResult;
+    assert.equal(successResult.absoluteBaseUrl, "/baz");
```
Comment (aleclarson) [inline src/__tests__/config-loader.test.ts:114]: Oops, this is using `assert` which is undefined.

@jonaskello Would be nice to run test CI on PRs!

Comment (nwalters512) [reply src/__tests__/config-loader.test.ts:114]: Good catch! Looks like my PR predated the move to Jest in https://github.com/dividab/tsconfig-paths/pull/186. I can patch up this test if you aren't already working on that.

Comment (aleclarson) [reply src/__tests__/config-loader.test.ts:114]: I'll let you handle it, thanks!

Comment (jonaskello) [reply src/__tests__/config-loader.test.ts:114]: @aleclarson Yes, definitely would be nice to run the tests in CI :-). It used to work but perhaps it was broken by the move to github actions. I'll see if I can get that running again.

Comment (nwalters512) [reply src/__tests__/config-loader.test.ts:114]: PR with passing test suite: https://github.com/dividab/tsconfig-paths/pull/211

**Pull request merged**